lib/rubocop/cop/rspec/match_array.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Prefer `contain_exactly` when matching an array literal.
+      #
+      # @example
+      #   # bad
+      #   it { is_expected.to match_array([content1, content2]) }
+      #
+      #   # good
+      #   it { is_expected.to contain_exactly(content1, content2) }
+      #
+      #   # good
+      #   it { is_expected.to match_array([content] + array) }
+      #
+      #   # good
+      #   it { is_expected.to match_array(%w(tremble in fear foolish mortals)) }
+      class MatchArray < Base
+        extend AutoCorrector
+
+        MSG = 'Prefer `contain_exactly` when matching an array literal.'
+
+        def on_send(node)
+          return unless node.method?(:match_array)
```

Comment:
I'm uncertain if we can rely on `RESTRIcT_ON_SEND` to replace this check. But I'll happily add it for performance.

Reply:
Simplecov complains about this line. But I disagree with the need to test it.

![image](https://user-images.githubusercontent.com/6916/214156362-a3c174cf-df5c-45af-9891-c57420d64f9b.png)

Adjusted the constraints.

Reply:
I don't get why the Test coverage job fails, as for me locally it's exactly within limits:
```
All Files ( 99.89% covered at 219.07 hits/line )
144 files in total.
3548 relevant lines, 3544 lines covered and 4 lines missed. ( 99.89% )
747 total branches, 712 branches covered and 35 branches missed. ( 95.31% ) 
```
@ydah WDYT?

Reply:
Sorry for the late reply. Looks like it's resolved.

Reply:
Yes, I think we do not need that line